### DISCUSSION TITLE
Syntax von "onClick" in React Aufgabe

### DISCUSSION THREAD

**Discussion by B-Baumeister** [neuefische/questions#691] [Web: Self-directed learning]
Hallo zusammen, 
ich sitze gerade an der Wiederholung einer React-Aufgabe und verstehe die Syntax hier nicht.
Welche Aufgabe?
**REACT STATE 1 - Aufgabe Counter**

mir geht es nur um folgenden Part: 

```javascript

``` import React from "react";
import "./styles.css";
import { useState } from "react";

export default function App() {
  const [count, setCount] = useState(0);

  return (
<div className="container">
     <p>{count}</p>
      <div className="button-container">
        <button type="button" onClick={() => {setCount(count - 1)}}>     <---- warum??
          -
        </button>
        <button type="button" onClick={() => {setCount(count + 1)}}>  <---- warum??
          +
        </button>
      </div>
    </div>
  );
}`

ich verstehe nicht warum man das hier:

onClick={() => {setCount(count - 1)}}

 so schreiben muss mit der schweiften Klammer, dann einfache Klammern und dann den Arror Pfeil. Kann mir das jemand erklären?
LG Bonnie

**Comment by ninaassmann** [comment]
wenn man in einem onClick direkt die Funktion schreibt und etwas übergeben möchte (hier Count - 1) zum Beispiel muss man das als Arrow Function schreibt

deswegen () => {setCount(count-1)}
ich glaube, die geschweiften Klammern um setCount... könnte man weg lassen, kann aber sein, dass prettier oder so die automatisch wieder setzt

**Reply by B-Baumeister** [reply]
ah okay.. dann füge ich das jetzt mal meiner wichtigen Info-Liste zu =) vielen dank =)

**Reply by F-Kirchhoff** [reply]
Die Klammern müssen da sein, siehe mein Kommentar. Ansonsten hat Nina vollkommen recht.

**Comment by F-Kirchhoff** [comment]
In React muss man immer zwischen zwei Situationen unterscheiden, in denen sich der Code befinden kann: 
- man ist in einem normalen JavaScript Block
- man ist in einem "JSX" Block (sieht aus wie HTML)

Wenn man gerade JSX schreibt, z.B. `<button onClick= >` und man möchte irgendeinen Wert oder Expression in Javascript einfügen (z.B. eine Zahl, ein Array, ein Object oder eine `Funktion`) dann muss man das ankündigen mit `geschweiften Klammern`. Damit öffnet man ein `Javascript Fenster` in dem nun Javascript geschrieben werden kann:

```js 
<button onClick={ function(){...} } type='button'>
----------------  ----------------  --------------
         |              |                |
        JSX        Javascript           JSX
```

Das was du dann als Wert bei onClick definierst ist eine `Arrow function`. Die schreibt man so:
```js
() => setCount(count + 1)
```

Zusammen ergibt das dann den Code den du gepostet hast.
